### PR TITLE
Generate French Battle Only button

### PR DESCRIPTION
**Pull request by zenseii**
This generates a Battle Only button if French resources are available:
image

Relates to this:
https://github.com/ihhub/fheroes2/issues/5114

I tried to center and align as good as possible - The box is an odd number so a perfect centering is not possible.
I don't want to bother you since I know you're quite busy, but if you should have the time to give a quick glance with your talent for spotting at the above image @LeHerosInconnu , then I would be very grateful! Don't mind that the buttons themselves are not aligned specifically in that image, that is just a quick cut and paste of them - the text is what matters.

### CODE REVIEW

**Comment by zenseii** [comment]
Hi @ihhub ! The button fonts are clean and I tried to be as greedy as possible with the generation of it, but if you see need for further optimization and ways to optimize it then I'm all ears. Also this font has 4 "layers" (I wish it had just 2 like German") which means I had to use some tricks like multiplication for the 'T' to get it clean and properly generated.

Also there is some slight dithering in the deepest/brightest layer for the 'Mode' part and for the 'AI' part. Unless there's a magic trick to uniform these colors I don't really see any need to put effort into fixing that.

File: src/fheroes2/agg/agg_image.cpp [hunk 1]
```diff
@@ -2013,6 +2013,55 @@ namespace fheroes2
...
+            switch ( id ) {
+            case ICN::BTNBATTLEONLY:
+                _icnVsSprite[id].resize( 2 );
+                for ( int32_t i = 0; i < static_cast<int32_t>( _icnVsSprite[id].size() ); ++i ) {
+                    Sprite & out = _icnVsSprite[id][i];
+                    out = GetICN( ICN::BTNNEWGM, 6 + i );
+                    // Clean the button
+                    Image uniform( 88, 23 );
+                    uniform.fill( ( i == 0 ) ? GetColorId( 216, 184, 152 ) : GetColorId( 184, 136, 96 ) );
+                    Copy( uniform, 0, 0, out, 25, 18, uniform.width(), uniform.height() );
```
Comment (ihhub) [inline src/fheroes2/agg/agg_image.cpp:2034]: We can use something like this:
```cpp
const uint8_t fillColor = ( i == 0 ) ? GetColorId( 216, 184, 152 ) : GetColorId( 184, 136, 96 );
Fill( out, 25, 18, 88, 23, fillColor );
```
At the time of initial button creation (it was actually changing the old image generation code) Fill function didn't exist.

This solution is much faster and easier to read. Could you please also do this for German and English?

File: src/fheroes2/agg/agg_image.cpp [hunk 2]
```diff
@@ -2013,6 +2013,55 @@ namespace fheroes2
                     Blit( GetICN( ICN::BTNNEWGM, 4 + i ), 36 - i, 23, out, 78 - i, 23, 10, 14 );
                 }
                 break;
+
+            default:
+                // You're calling this function for non-specified ICN id. Check your logic!
+                assert( 0 );
+                break;
+            }
+        }
```
Comment (ihhub) [inline src/fheroes2/agg/agg_image.cpp:2022]: Please add an empty line to separate 2 functions.

File: src/fheroes2/agg/agg_image.cpp [hunk 3]
```diff
@@ -2013,6 +2013,55 @@ namespace fheroes2
...
+                    // Add 'T'
+                    Blit( GetICN( ICN::BTNDC, 2 + i ), 89 - i, 21, out, 50 - i, secondLine, 12, 15 );
+                    // Clean up 'AT'
+                    Blit( GetICN( ICN::BTNDC, 2 + i ), 89 - i, 18, out, 50 - i, secondLine, 1, 1 );
+                    Blit( GetICN( ICN::BTNDC, 2 + i ), 92 - ( 5 * i ), 27 - i, out, 49 - i, secondLine + 4 + i, 1, 3 );
+                    // Add 'AI'.
+                    Blit( GetICN( ICN::BTNMP, 6 + i ), 56 - i, 13, out, 62 - i, secondLine, 18, 15 );
+                    // Clean up 'TA'
+                    Blit( GetICN( ICN::BTNBAUD, 2 + i ), 51 - i, 40, out, 60 - i, secondLine + 12, 3, 3 );
+                    // add 'LLE'
```
Comment (ihhub) [inline src/fheroes2/agg/agg_image.cpp:2054]: To be consistent: please make **a** as a capital letter.

**Review by ihhub — CHANGES_REQUESTED** [review]
Hi @zenseii , I put few comments here. Would you mind to take a look at them?

File: src/fheroes2/agg/agg_image.cpp (hunk 1 above)
Comment (zenseii) [reply src/fheroes2/agg/agg_image.cpp:2034]: Sure, I was actually wondering why Fill() was not used.

**Review by ihhub — APPROVED** [review]

**Comment by ihhub** [comment]
@zenseii , a lot of thanks for working on translations!

**Pull request merged**